crates/viewer/re_viewer_context/src/component_fallbacks.rs

```diff
@@ -37,87 +48,241 @@ pub enum ComponentFallbackError {
-    ) -> ComponentFallbackProviderResult;
+        provider: ComponentFallbackProviderFn,
+    ) {
+        if self
+            .type_fallback_providers
+            .insert(component, provider)
+            .is_some()
+        {
+            re_log::warn!(
+                "There was already a component fallback provider registered for {component}"
+            );
+        }
+    }
+
+    /// Registers a fallback provider function for a given component type.
+    pub fn register_type_fallback_provider<C: re_types::Component>(
+        &mut self,
+        f: impl Fn(&QueryContext<'_>) -> C + Send + Sync + 'static,
+    ) {
+        self.register_dyn_type_fallback_provider(
+            C::name(),
+            Box::new(move |query_context| {
+                let value = f(query_context);
+
+                C::to_arrow([Cow::Owned(value)])
+            }),
+        );
+    }
+
+    /// Registers a fallback provider function for a given component identifier.
+    pub fn register_dyn_fallback_provider(
+        &mut self,
+        identifier: ComponentIdentifier,
+        provider: ComponentFallbackProviderFn,
+    ) {
+        if self
+            .component_fallback_providers
+            .insert(identifier, provider)
+            .is_some()
+        {
+            re_log::warn!(
+                "There was already a component fallback provider registered for {identifier}"
+            );
+        }
+    }
+
+    /// Registers a fallback provider function for a given component identifier.
+    pub fn register_fallback_provider<C: re_types::Component>(
+        &mut self,
+        descriptor: &ComponentDescriptor,
+        provider: impl Fn(&QueryContext<'_>) -> C + Send + Sync + 'static,
+    ) {
+        debug_assert_eq!(descriptor.component_type, Some(C::name()));
+
+        self.register_dyn_fallback_provider(
+            descriptor.component,
+            Box::new(move |query_context| {
+                let value = provider(query_context);
+
+                C::to_arrow([Cow::Owned(value)])
+            }),
+        );
+    }
+
+    /// Registers a fallback provider function for a given component identifier
+    /// in a specific view.
+    pub fn register_dyn_view_fallback_provider(
+        &mut self,
+        view: ViewClassIdentifier,
+        component: ComponentIdentifier,
+        provider: ComponentFallbackProviderFn,
+    ) {
+        if self
+            .view_component_fallback_providers
+            .insert((view, component), provider)
+            .is_some()
+        {
+            re_log::warn!(
+                "There was already a view component fallback provider registered for {component} in {view}"
+            );
+        }
+    }
+
+    /// Registers a fallback provider function for a given component identifier
+    /// in a specific view.
+    pub fn register_view_fallback_provider<C: re_types::Component>(
+        &mut self,
+        view: ViewClassIdentifier,
+        descriptor: &ComponentDescriptor,
```

Comment:
as above, let's not take descriptors when we don't index over them to avoid confusion
Yes, it's unfortunate that we can't assert on the type anymore then but I had to remove that on various type converting queries as well recently; no turning back and I think it's the lesser evil